src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -218,10 +216,27 @@ Json::Value SotaUptaneClient::AssembleManifest() {
   version_manifest[primary_ecu_serial.ToString()] = uptane_manifest->sign(primary_manifest, report_counter);
 
   for (auto it = secondaries.begin(); it != secondaries.end(); it++) {
+    const Uptane::EcuSerial &ecu_serial = it->first;
     Uptane::Manifest secmanifest = it->second->getManifest();
 
+    bool from_cache = false;
+    if (secmanifest == Json::Value()) {
+      // could not get the secondary manifest, a cached value will be provided
+      std::string cached;
+      if (storage->loadCachedEcuManifest(ecu_serial, &cached)) {
```

Comment:
This 'manifest caching' functionality looks like IP Secondary specific functionality, so I would 'hide' it behind the secondary interface and  implemented it in `IpUptaneSecondary::getManifest()`.

Reply:
I would like that also, but there should then be a way to access the storage from there. To evaluate

Reply:
Maybe dedicated table in sql.db or even DB that is initialized in IP secondary factory (preferable) or ctor.